seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkWriter.java

```diff
@@ -75,16 +87,17 @@ public void write(SeaTunnelRow element) throws IOException {
                             .indexOf(this.option.getShardMetadata().getShardKey());
             shardKey = element.getField(i);
         }
-        ClickhouseBatchStatement statement = statementMap.get(shardRouter.getShard(shardKey));
-        JdbcBatchStatementExecutor clickHouseStatement = statement.getJdbcBatchStatementExecutor();
-        IntHolder sizeHolder = statement.getIntHolder();
+        this.statement = statementMap.get(shardRouter.getShard(shardKey));
+        this.clickHouseStatement = statement.getJdbcBatchStatementExecutor();
+        this.sizeHolder = statement.getIntHolder();
```

Comment:
Don't use `this` to save fields which mapping with shardKey, because one writer can hold many different shardKey, so it will have many `statement`,`sizeHolder`. Use `this` to save it may cause conflusion.

Reply:
Should it be？
`statement = statementMap.get(shardRouter.getShard(shardKey));
clickHouseStatement = statement.getJdbcBatchStatementExecutor();
sizeHolder = statement.getIntHolder();
`

Reply:
![image](https://user-images.githubusercontent.com/76689593/224617682-17b2078e-7efe-4af6-aab8-a8f5ae6bdba8.png)

Reply:
> Should it be？ `statement = statementMap.get(shardRouter.getShard(shardKey));
> 
> clickHouseStatement = statement.getJdbcBatchStatementExecutor();
> 
> sizeHolder = statement.getIntHolder(); `

No, just do not use `ClickhouseSinkWriter` object's fields to save `statement` and  `sizeHolder`. Because it belong shard, not writer. One writer may have many shard.

Reply:
****

> > Should it be？ `statement = statementMap.get(shardRouter.getShard(shardKey));
> > clickHouseStatement = statement.getJdbcBatchStatementExecutor();
> > sizeHolder = statement.getIntHolder(); `
> 
> No, just do not use `ClickhouseSinkWriter` object's fields to save `statement` and `sizeHolder`. Because it belong shard, not writer. One writer may have many shard.

Can you do some hints on code?

Reply:
![Uploading image.png…]()

Reply:
![image](https://user-images.githubusercontent.com/76689593/224621028-1cf36942-7947-4b57-9692-e71bc34f484f.png)
**statement** and **sizeHolder**   Create internal objects?

Reply:
I mean one writer may have many `sizeHolder`, `clickHouseStatement` and `statement`. Because one writer have many shard, one shard have one `sizeHolder`, `clickHouseStatement` and `statement`. So the `this.sizeHolder` will change when last row had change ( the row change then the last shard also change).

Reply:
Thank you for your explanation. I think I understand